AmplifyPlugins/Predictions/CoreMLPredictionsPlugin/Dependency/CoreMLSpeechAdapter.swift

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright 2018-2019 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Speech
+
+class CoreMLSpeechAdapter: CoreMLSpeechBehavior {
+
+    func getTranscription(_ audioData: URL) -> SpeechToTextResult? {
+        let request = SFSpeechURLRecognitionRequest(url: audioData)
+        request.requiresOnDeviceRecognition = true
+        var transcriptions = [String]()
+        let recognizer = SFSpeechRecognizer()
+        let semaphore = DispatchSemaphore(value: 0)
```

Comment:
Can we do this without semaphores and return the result in a callback?

Reply:
done